src/WindowsStateTriggers/CompositableAdaptiveTrigger.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Core;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+
+namespace WindowsStateTriggers
+{
+    /// <summary>
+    /// Extends the <see cref="AdaptiveTrigger"/> functionality with 
+    /// <see cref="ITriggerValue"/> interface implementation 
+    /// for <see cref="CompositeStateTrigger"/> usage
+    /// </summary>
+    public class CompositableAdaptiveTrigger : AdaptiveTrigger, ITriggerValue
+    {
+        /// <summary>
+		/// Initializes a new instance of the <see cref="FullScreenModeTrigger"/> class.
+		/// </summary>
+        public CompositableAdaptiveTrigger()
+        {
+            this.RegisterPropertyChangedCallback(MinWindowHeightProperty, OnMinWindowHeightPropertyChanged);
+            this.RegisterPropertyChangedCallback(MinWindowWidthProperty, OnMinWindowWidthPropertyChanged);
+
+            CoreApplication.GetCurrentView().CoreWindow.SizeChanged += OnCoreWindowOnSizeChanged;
```

Comment:
This must be a weak eventlistener to avoid leaking. Also CurrentView and CoreWindow can return null.